parser/parser.go

```diff
@@ -104,3 +104,36 @@ func packageName(f *ast.File) string {
 	}
 	return defaultPackageName
 }
+
+// MergeExternToSemanticHandle takes an extern and a parsed query, creates a package with the extern,
+// converts both AST packages to libflux.ASTPkg types, merges those packages and analyzes them, returning
+// a libflux.SemanticPkg
+func MergeExternToSemanticHandle(extern *ast.File, sourcePkg *ast.Package) (*libflux.SemanticPkg, error) {
+	var externList []*ast.File
+	externList = append(externList, extern)
+	externPkg := &ast.Package{
+		Files:    externList,
```

Comment:
This can be simplified to `Files: []*ast.File{extern}` and the `append` logic removed.